staticcheck/lint.go

```diff
@@ -3926,20 +3923,28 @@ func checkJSONTag(pass *analysis.Pass, field *ast.Field, tag string) {
 					}
 				}
 			}
-		case "inline":
-			ci++
+			options[s] = options[s] + 1
+		case "omitzero", "omitempty", "nocase", "inline", "unknown":
+			options[s] = options[s] + 1
 		default:
+			if strings.HasPrefix(s, "format:") {
+				options["format"] = options["format"] + 1
+				break
+			}
 			report.Report(pass, field.Tag, fmt.Sprintf("unknown JSON option %q", s))
 		}
 	}
-	if co > 1 {
-		report.Report(pass, field.Tag, `duplicate JSON option "omitempty"`)
-	}
-	if cs > 1 {
-		report.Report(pass, field.Tag, `duplicate JSON option "string"`)
+	var duplicates []string
+	for option, n := range options {
+		if n > 1 {
+			duplicates = append(duplicates, option)
+		}
 	}
-	if ci > 1 {
-		report.Report(pass, field.Tag, `duplicate JSON option "inline"`)
+	if len(duplicates) > 0 {
+		sort.Strings(duplicates)
+		for _, option := range duplicates {
+			report.Report(pass, field.Tag, `duplicate JSON option `+strconv.Quote(option))
```

Comment:
Let's use fmt.Sprintf and %q here, for consistency with the rest of the codebase

Reply:
Done.